app/controllers/quotes_controller.rb

```diff
@@ -0,0 +1,7 @@
+class QuotesController < ApplicationController
+
+  def search
+    @quotes_info = QuotesFacade.searched_quotes(params[:query])
+    @quotes = QuotesFacade.make_objects(@quotes_info)
+  end
+end
```

Comment:
This is okay, and will work. But the thing I don't love about it is that you're sending ALL the data from the api call to your view. So, I'd update this to be something like: 

```ruby
    quotes_info = QuotesFacade.searched_quotes(params[:query])
    @total_count = quotes_info[:totalCount]
    @quotes = QuotesFacade.make_objects(quotes_info)
```

Reply:
You could also send two things back from the searched_quotes method. To do that, you'd do something like this in your facade: 

```ruby
  def self.searched_quotes(query)
    quotes_info = QuoteService.search_quotes_api_call(query)
    quotes = quotes[:results].map do |quote|
      QuoteObject.new(quote)
    end
    total_count, quotes = [quotes_info[:totalCount], quotes]
  end
```

and then in your controller: 

```ruby
 def search
    @total_count, @quotes = QuotesFacade.searched_quotes(params[:query])
end 
```